Bourreau/lib/bourreau_system_checks.rb

```diff
@@ -289,15 +289,22 @@ def self.a100_ensure_dp_cache_symlink_exists #:nodoc:
     myself        = RemoteResource.current_resource
     gridshare_dir = myself.cms_shared_dir
     cache_dir     = myself.dp_cache_dir
+    sym_path      = "#{gridshare_dir}/#{DataProvider::DP_CACHE_SYML}"
 
     return unless Dir.exists?(gridshare_dir) && Dir.exists?(cache_dir)
 
     #----------------------------------------------------------------------------
     puts "C> Making sure the grid share directory has a symlink to the data provider cache..."
     #----------------------------------------------------------------------------
 
-    sym_path = "#{gridshare_dir}/#{DataProvider::DP_CACHE_SYML}"
-    return if File.symlink?(sym_path) && File.realpath(sym_path) == File.realpath(cache_dir)
+    # update the work directory timestamp to counter cluster deletion policies
```

Comment:
Why is this so complicated?!??

The original code's logic was:

```
return if symlink_is_ok
create symlink
```

The new code should simply be instead:

```
if symlink_is_ok
  touch symlink
  return
end
create symlink
```

But you did a complex set of conditions including checking the return status of system():

```
touch_but_not_create symlink
ret=system(touch symlink)
return if ret is 0 and symlink_is_ok
create_symlink
```

Why is make it so complicated?

Reply:
To make code work even if touch fails. But ok, can delete.